concordium-contracts-common/src/schema.rs

```diff
@@ -229,6 +252,27 @@ impl FunctionV2 {
     }
 }
 
+/// Describes the schema of an init or a receive function for V1 contracts with
+/// V3 schemas. Differs from [`FunctionV2`] in that a schema for the events can
+/// be included.
```

Comment:
I think the second sentence should be removed now. It is no longer true. We could also just have FunctionV2 instead of this type seeing that they are the same. And then use FunctionV2 in the `init` and `receive` parts of `ContractV3`.